cardano-cli/src/Cardano/CLI/Run/Legacy/Transaction.hs

```diff
@@ -1189,24 +1192,28 @@ runTxSubmit socketPath (AnyConsensusModeParams cModeParams) network txFilePath =
 -- Transaction fee calculation
 --
 
-runTxCalculateMinFee
-  :: TxBodyFile In
+runTxCalculateMinFee :: ()
+  => AnyCardanoEra
+  -> TxBodyFile In
   -> NetworkId
   -> ProtocolParamsFile
   -> TxInCount
   -> TxOutCount
   -> TxShelleyWitnessCount
   -> TxByronWitnessCount
   -> ExceptT ShelleyTxCmdError IO ()
-runTxCalculateMinFee (File txbodyFilePath) nw pParamsFile
-                     (TxInCount nInputs) (TxOutCount nOutputs)
-                     (TxShelleyWitnessCount nShelleyKeyWitnesses)
-                     (TxByronWitnessCount nByronKeyWitnesses) = do
+runTxCalculateMinFee
+      anyEra
+      (File txbodyFilePath) nw pParamsFile
+      (TxInCount nInputs) (TxOutCount nOutputs)
+      (TxShelleyWitnessCount nShelleyKeyWitnesses)
+      (TxByronWitnessCount nByronKeyWitnesses) = do
+    AnyCardanoEra era <- pure anyEra
 
     txbodyFile <- liftIO $ fileOrPipe txbodyFilePath
     unwitnessed <- firstExceptT ShelleyTxCmdCddlError . newExceptT
                      $ readFileTxBody txbodyFile
-    pparams <- firstExceptT ShelleyTxCmdProtocolParamsError $ readProtocolParameters pParamsFile
+    pparams <- firstExceptT ShelleyTxCmdProtocolParamsError $ readProtocolParameters era pParamsFile
```

Comment:
The reason the `era` is necessary is because the process of reading the protocol parameters which happens inside this function is era-sensitive.